definitions/npm/@reach/router_v1.1.x/flow_v0.63.x-/router_v1.1.x.js

```diff
@@ -49,66 +56,77 @@ declare module '@reach/router' {
   };
 
   declare export type LocationProviderRenderFn = (context: {|
-    location: typeof location,
+    location: DomLocation,
     navigate: NavigateFn,
   |}) => React$Node;
 
-  declare export class Router extends React$Component<{|
+  declare export var Router: React$StatelessFunctionalComponent<{|
     children?: React$Node,
     basepath?: string,
     primary?: boolean,
-    location?: typeof location,
-  |}> {}
+    location?: DomLocation,
+  |}>;
 
-  declare export class Link<State> extends React$Component<{
+  declare export type LinkProps<State> = {
     ...$Shape<HTMLAnchorElement>,
     children: React$Node,
+    innerRef?: React$Ref<HTMLAnchorElement>,
     getProps?: (props: {
       isCurrent: boolean,
       isPartiallyCurrent: boolean,
       href: string,
-      location: typeof location,
+      location: DomLocation,
     }) => {},
     state?: State,
     to?: string,
     replace?: boolean,
     href?: empty, // remove href, as it's ignored by the router
-  }> {}
+  };
+
+  declare export var Link: {
+    <State>(props: LinkProps<State>): React$Node,
```

Comment:
why not just function instead of callable object?

Reply:
I had a [good reason](https://flow.org/try/#0JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wG4AocgEyTQBsUi5MBXAOw2AjbgEF+AFAG80AC2B0qRNgC44AgJRwAvAD44ANwjAqAXwVzk6GABIAchBoVqtBkw2M4AIRcyh5AJDCxEqUlnySmqa2noGiMSmFlbkupTkADz8vKoAzjBQwGwA5gkA9MmqpHB5eXAA7qJITNU4UHAA-HAAYuyc3HA09IwoMFw8LKlIdEipqZQJLk5pGVm5eVNFQA)

Reply:
You forgot about generic in this case
`<State>(props: LinkProps<State>): React$Node,`

Reply:
True! Asked about this in the issue you've made: https://github.com/facebook/flow/issues/7672#issuecomment-488687715

Reply:
flow.js does not support generic types for variables. it makes no sense now

Reply:
Callable object is more suitable as a component can be include `defaultProps` and `propTypes`
```
StatelessFunctionalComponent === {
    (Props): React$Node.
     defaultProps?: {},
     propTypes?: {},
}
```

Reply:
Very good point! Though, it's only more suitable if it really includes them :)

Reply:
Well, as we see, the usual functions do not perceive the flow as StatelessFunctionComponent

https://github.com/flow-typed/flow-typed/pull/3281#discussion_r280302543